src/data/device_automation.ts

```diff
@@ -0,0 +1,39 @@
+import { HomeAssistant } from "../types";
+
+export interface DeviceTrigger {
+  platform: string;
+  device_id: string;
+  domain: string;
+  entity_id: string;
+  type?: string;
+  event?: string;
+}
+
+export interface DeviceTriggerList {
+  triggers: DeviceTrigger[];
+}
+
+export const fetchDeviceTriggers = (hass: HomeAssistant, deviceId: string) =>
+  hass.callWS<DeviceTriggerList>({
+    type: "device_automation/trigger/list",
+    device_id: deviceId,
+  });
+
+export const triggersEqual = (a: any, b: any) => {
```

Comment:
Easier to just set this type to `DeviceTrigger`
```suggestion
export const triggersEqual = (a: DeviceTrigger, b: DeviceTrigger) => {
```

Reply:
No need to make a very generic equal function, just type it and we can restrict it.